presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestingArrowMetadata.java

```diff
@@ -0,0 +1,157 @@
+    {
+        super(config, clientHandler);
+        this.nodeManager = nodeManager;
+        this.testconfig = testconfig;
+        this.clientHandler = clientHandler;
+        this.config = config;
+    }
+
+    @Override
+    protected ArrowFlightRequest getArrowFlightRequest(ArrowFlightConfig config, String schema)
+    {
+        return new TestingArrowFlightRequest(config, schema, nodeManager.getWorkerNodes().size(), testconfig);
+    }
+
+    @Override
+    public List<String> listSchemaNames(ConnectorSession session)
+    {
+        List<String> listSchemas = extractSchemaAndTableData(Optional.empty(), session);
+        List<String> names = new ArrayList<>();
+        for (String value : listSchemas) {
+            names.add(value.toLowerCase(ENGLISH));
+        }
+        return ImmutableList.copyOf(names);
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaName)
+    {
+        String schemaValue = schemaName.orElse("");
+        String dataSourceSpecificSchemaName = getDataSourceSpecificSchemaName(config, schemaValue);
+        List<String> listTables = extractSchemaAndTableData(Optional.ofNullable(dataSourceSpecificSchemaName), session);
+        List<SchemaTableName> tables = new ArrayList<>();
+        for (String value : listTables) {
+            tables.add(new SchemaTableName(dataSourceSpecificSchemaName.toLowerCase(ENGLISH), value.toLowerCase(ENGLISH)));
+        }
+
+        return tables;
+    }
+
+    public List<String> extractSchemaAndTableData(Optional<String> schema, ConnectorSession connectorSession)
+    {
+        try (ArrowFlightClient client = clientHandler.getClient(Optional.empty())) {
+            List<String> names = new ArrayList<>();
+            ArrowFlightRequest request = getArrowFlightRequest(config, schema.orElse(null));
+            ObjectNode rootNode = (ObjectNode) objectMapper.readTree(request.getCommand());
+
+            String modifiedQueryJson = objectMapper.writeValueAsString(rootNode);
+            byte[] queryJsonBytes = modifiedQueryJson.getBytes(StandardCharsets.UTF_8);
+            Iterator<Result> iterator = client.getFlightClient().doAction(new Action("discovery", queryJsonBytes), clientHandler.getCallOptions(connectorSession));
+            while (iterator.hasNext()) {
+                Result result = iterator.next();
+                String jsonResult = new String(result.getBody(), StandardCharsets.UTF_8);
+                List<String> tableNames = objectMapper.readValue(jsonResult, new TypeReference<List<String>>() {
+                });
+                names.addAll(tableNames);
+            }
+            return names;
+        }
+        catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    protected Type overrideFieldType(Field field, Type type)
```

Comment:
Instead of having an `overideFieldType`, can we instead provide a method which simply maps the Arrow type to the Presto type, and this can be overridden or enriched?  I.e. just make the whole mapping configurable.

Reply:
Instead of this method we are invoking `getPrestoTypeFromArrowField` which can be overridden